### PR TITLE
fix wasm, update changelog

### PR DESCRIPTION
**Pull request by cqc-melf**
fixes the issues of the wasm parsing on the pytket side.

I would like to put out a release candidate after this is merged

### CODE REVIEW

File: pytket/pytket/wasm/wasm.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ def __init__(self, filepath: str, check_file: bool = True, int_size: int = 32):
 
         function_signatures: list = []
         function_names: list = []
+        _func_lookup = {}
```
Comment (cqc-alec) [inline pytket/pytket/wasm/wasm.py:71]: Can we please add a test that shows the issue this is fixing (i.e. fails before the fix)?

Comment (cqc-melf) [reply pytket/pytket/wasm/wasm.py:71]: I have added one additional file that is now read in

File: pytket/tests/classical_test.py [hunk 2]
```diff
@@ -458,6 +458,52 @@ def test_wasmfilehandler_repr_64() -> None:
     )
 
 
+def test_wasmfilehandler_repr_2() -> None:
```
Comment (cqc-alec) [inline pytket/tests/classical_test.py:461]: Looks like this test (and the following one) pass with the existing version, or am I missing something? Do they exhibit the bug?

File: pytket/tests/wasm-generation/src/lib-2.rs [hunk 3]
```diff
@@ -0,0 +1,107 @@
+// src/lib.rs
+
+#[no_mangle]
+fn init() {
```
Comment (cqc-melf) [inline pytket/tests/wasm-generation/src/lib-2.rs:4]: for example here

File: pytket/tests/wasm-generation/src/lib-2.rs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    let mut i = 0;
+
+    while i < limit {
+        i = i * 2;
+    }
+return i
+}
+
+#[no_mangle]
+pub fn mixed_up_2(limit: i32, limit2: i32) -> i32 {
```
Comment (cqc-melf) [inline pytket/tests/wasm-generation/src/lib-2.rs:65]: and here

File: pytket/tests/wasm-generation/src/lib-2.rs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+    while i < limit2 {
+        i = i * 3;
+    }
+return i
+}
+
+
+#[no_mangle]
+fn mixed_up_3(limit: i32, limit2: i32, limit3: i32) -> i32 {
```
Comment (cqc-melf) [inline pytket/tests/wasm-generation/src/lib-2.rs:80]: and here

File: pytket/tests/classical_test.py (hunk 2 above)
Comment (cqc-alec) [reply pytket/tests/classical_test.py:461]: Ah OK, thanks!

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**